crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -99,6 +234,25 @@ impl Inner {
         }
     }
 
+    async fn blocking_estimate_prices_and_update_cache(
+        &self,
+        token: &H160,
+        max_age: Duration,
+    ) -> Result<Option<f64>, PriceEstimationError> {
+        if let Some(batching) = self.batching.as_ref() {
```

Comment:
The batching should be an optional feature, we don't necessarily want it for the autopilot, but we do for the orderbook. Nevertheless, once we move everything to its own service this will be heavily improved.